### PR TITLE
[test_rom] enable icache / Ibex security features

### LINKED ISSUE #12883: [sw] Enable Ibex's icache in test_rom
The icache within Ibex starts disabled and needs to be specifically enabled via a write to the Ibex specific `cpuctrl` CSR (see https://ibex-core.readthedocs.io/en/latest/03_reference/cs_registers.html#cpu-control-register-cpuctrl).

Do we enable the icache anywhere? We should turn it on for at least some chip level testing (as these are also helping verify Ibex along with the specific things they're aimed at). Has any consideration be given as to when to enable it in the boot flow?

@arunthomas @cfrantz @mundaym

### PR DESCRIPTION
**Pull request by timothytrippel**
This updates the test ROM to enable the icache (and other cpuctrl controlled Ibex security features) based on the OTP-stored configuration word.

This fixes #12883.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Review by alphan — APPROVED** [review]

File: sw/device/lib/testing/test_rom/test_rom.c [hunk 1]
```diff
@@ -83,6 +84,16 @@ bool rom_test_main(void) {
...
+  // Initialize Ibex cpuctrl (contains icache / security feature enablements).
+  uint32_t cpuctrl_csr;
+  CSR_READ(CSR_REG_CPUCTRL, &cpuctrl_csr);
+  uint32_t cpuctrl_otp_val = abs_mmio_read32(
+      TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET +
+      OTP_CTRL_PARAM_CREATOR_SW_CFG_CPUCTRL_OFFSET);
+  cpuctrl_csr = bitfield_field32_write(
+      cpuctrl_csr, (bitfield_field32_t){.mask = 0x3f, .index = 0},
+      cpuctrl_otp_val);
+  CSR_WRITE(CSR_REG_CPUCTRL, cpuctrl_csr);
```
Comment (alphan) [inline sw/device/lib/testing/test_rom/test_rom.c:96]: I think this should go in the `ifdef` block above or english breakfast will be sad (it doesn't have OTP IIRC).

**Review by alphan — APPROVED** [review]

File: sw/device/lib/testing/test_rom/test_rom.c (hunk 1 above)
Comment (timothytrippel) [reply sw/device/lib/testing/test_rom/test_rom.c:96]: oops, yes this was supposed to go in the block above. thanks for the reminder! fixed.

**Comment by tjaychen** [comment]
should we update the default otp images to turn on icache? Or maybe that's already done.

**Comment by alphan** [comment]
> should we update the default otp images to turn on icache? Or maybe that's already done.

Yes, it was already enabled for RMA. I think after the last refactoring it's enabled for every lifecycle state. See [here](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/otp_ctrl/data/otp_ctrl_img_creator_sw_cfg.hjson;l=57).

**Comment by tjaychen** [comment]
thanks @alphan

**Pull request merged**